zerver/webhooks/dialogflow/view.py

```diff
@@ -0,0 +1,35 @@
+# Webhooks for external integrations.
+from typing import Text, Any, Dict
+from django.http import HttpRequest, HttpResponse
+from zerver.decorator import api_key_only_webhook_view
+from zerver.lib.actions import check_send_private_message
+from zerver.lib.request import REQ, has_request_variables
+from zerver.lib.response import json_success
+from zerver.models import UserProfile, get_user_profile_by_email
+
+@api_key_only_webhook_view("dialogflow")
+@has_request_variables
+def api_dialogflow_webhook(request: HttpRequest, user_profile: UserProfile,
+                           payload: Dict[str, Any]=REQ(argument_type='body'),
+                           email: str=REQ(default='foo')) -> HttpResponse:
+    user_query = payload["result"]["resolvedQuery"]
+    status = payload["status"]["code"]
+
+    if status == 200:
+        result = payload["result"]["fulfillment"]["speech"]
+        if result == '':
+            alternate_result = payload["alternateResult"]["fulfillment"]["speech"]
+            if alternate_result == '':
+                output = u"Sorry. I didn't get that."
+            else:
+                output = alternate_result
+        else:
+            output = result
+    else:
+        error_status = payload["status"]["errorDetails"]
+        output = u"{} - {}".format(status, error_status)
+
+    profile = get_user_profile_by_email(email)
+    body = u"You asked Dialogflow : {} \n Response : **{}**".format(user_query, output)
```

Comment:
I don't think we need to repeat what the user asked. The user can already see that, since its a private message conversation. I think body should just be equal to `output`. So, remove this line (line 33) and do:
```
check_send_private_message(user_profile, request.client, profile, output)
```